source/common/router/config_utility.cc

```diff
@@ -22,6 +22,8 @@ namespace Router {
 //   b.regex_match: Match will succeed if header value matches the value specified here.
 //   c.range_match: Match will succeed if header value lies within the range specified
 //     here, using half open interval semantics [start,end).
+// For a regex match the configured regex string is stored on the 'value_' member to aid
+// in debugging.
```

Comment:
Won't logging be sufficient? I think it's generally better to avoid adding debug stuff to release builds like this, despite the fact it's mostly harmless here.